### DISCUSSION TITLE
Bolt with 4 in 1 esc issue

### DISCUSSION THREAD

**Discussion by kimseong-yeon** [bitcraze/discussions#196] [Q&A] [answered]
Hi. 
I'm testing Bolt 1.1 with flow deck and 4in1 ESC with Dshot 600 
(ESC : http://www.tcmm-rc.com/product-center/fpv-4-in-1-30a-blheli-esc-s-2-5s-rc-racing-drone%e5%9b%9b%e6%97%8b%e7%a9%bf%e8%b6%8a%e6%9c%ba.html)

1. At first, with Cfclient, When i connect the battery(2S) to the ESC and bolt turned on, esc calibration beep starts and motor 1 or 2 turn periodically.
and I throttle up, there are motor delay which maybe being on calibration mode
if i throttled up continuously, motor thrusts well ( just hovering with 4 point string fixed )
and if i throttled up and down, some motors turn or not.
and when i throttle stop, periodical beep and turn of motor starts.
(https://youtube.com/shorts/N4YznUysLJI?feature=share)

I searched Dshot protocol doesn't need calibration.

can you let me know what is problem?
at first i thought it was because the ESC was not calibrated or Dshot protocol number was wrong,
but maybe these are not the problem.

2. and additionally sometimes motor spins abnormally like this (with flow deck)
(https://youtube.com/shorts/bpZSWXWawVU?feature=share)

sometimes IMU data on cfclient (flight control tab-screen) splash or drifting, and motor spins like over signal.
after i connect Flow deck, this problem became decreased but still there.
especially roll and pitch.

if it is because of accumulated error of IMU or PID, is that inevitable?

**Comment by kimseong-yeon** [comment]
i got it with system-forceArm configuration on menuconfig and parameter update

but i think there are some drift on IMU with take off script in step-by-step.
move slightly on xy direction

**Comment by knmcguire** [comment]
Hi! Sorry for the delay but just a headsup. This is a good question for @tobbeanton but he just came back from his holidays. He'll answer your question in the next few days.

**Comment by tobbeanton** [comment, marked as answer]
@kimseong-yeon 
Did you get the 4in1 to run reliably?

Regarding the drift, this is inevitable without some form of absolute positioning system.

**Reply by kimseong-yeon** [reply]
hi 
yes. i run it with disarming on start.
For drift, i would find other sensor or systems

thank you!!
